core/src/main/java/com/sequenceiq/cloudbreak/service/ServiceEndpointCollector.java

```diff
@@ -312,6 +318,28 @@ private void addImplaDebugUrl(String managerIp, Gateway gateway, Map<String, Lis
         }
     }
 
+    private void addKuduUrl(String managerIp, Gateway gateway, Map<String, List<String>> privateIps, boolean autoTlsEnabled, List<String> urls) {
+        Optional<String> coordinatorUrl = privateIps.get(ExposedService.KUDU.getServiceName())
+                .stream()
+                .map(coordinator -> getKuduUrl(gateway, managerIp, coordinator, autoTlsEnabled))
+                .flatMap(Optional::stream)
+                .findFirst();
+        if (coordinatorUrl.isPresent()) {
+            urls.add(coordinatorUrl.get());
+        }
+    }
+
+    private void addKuduTUrl(String managerIp, Gateway gateway, Map<String, List<String>> privateIps, boolean autoTlsEnabled, List<String> urls) {
+        Optional<String> coordinatorUrl = privateIps.get(ExposedService.KUDU_TABLET_SERVER.getServiceName())
+                .stream()
+                .map(coordinator -> getKuduTUrl(gateway, managerIp, coordinator, autoTlsEnabled))
+                .flatMap(Optional::stream)
+                .findFirst();
```

Comment:
This suggests the ability to publish a link to the tserver web UIs, and an arbitrary selection of the first tserver. I would recommend against that. Let's only publish links to the "first" master's web UI; all tserver web UIs are reachable from any master web UI.

Reply:
So we dont want to publish the tserver url at all ?

Reply:
I don’t think so. A typical Kudu cluster may have 3 or 5 masters and dozens (or more) tservers. Individual tserver web UIs aren’t that interesting with a cluster of that size, but the master is always compelling.

Moreover, there are links to the tserver web UIs (and to the other masters too) from inside each master’s web UI.

Reply:
removed that so the tablet server is not in the list